src/java.base/aix/native/libjava/ProcessHandleImpl_aix.c

```diff
@@ -36,9 +38,127 @@
 
 void os_initNative(JNIEnv *env, jclass clazz) {}
 
+/*
+ * Returns the children of the requested pid and optionally each parent and
+ * start time. If requested pid is zero return all processes.
```

Comment:
This deserves at least a better explanation, because I was first wondering why you return the parent pid if you hand in the parent pid, then I thought "oh, maybe he traverses indirect children too, then returning the respective parent pid would make sense", and then I saw that no, you only return direct children unless pid is 0.

So, from looking at the implementation, the only point of the parent pid return array is if you pass in 0 as pid. This should be made clearer. Or even better, split this into two functions, one "get children(pid, returnpidarray)" and one "getAllProcesses(returnarray, ppid returnarray)".

Reply:
Hi Thomas, this function is just the same as the linux and macos implementation. And I do not want to make more changes as really needed, otherwise the siblings character gets lost. Following your proposal means to do the split for linux and macos too. And to be honest, I do not really know the intention of the function, maybe the implementation is wrong on all platforms. So my goal was to only exchange the used API and keep everything else as on linux. This is my defensive approach.

Reply:
Probably keep the function without split into 2 functions, but just adjust the comment a bit  as Thomas suggested ?

Reply:
The callers in ProcessHandlerImpl depend on knowing the parent pid for each process. 
They are used to find the descendants of a process. When reading /proc that's necessary.

Reply:
@RogerRiggs Roger you are most familiar with os_getChildren(). How would you phrase the introducing comment of this function?